tensorrt_llm/_torch/autotuner.py

```
@@ -475,14 +476,25 @@ def _serialize_cache_to_json(self) -> Dict[str, Any]:
 
         for key, value in self.cache.items():
-            # Convert tuple key to string for JSON compatibility
+            # Convert any simple object to string for JSON compatibility
             key_str = str(key)
 
             runner_id, tactic, min_time = value
 
+            # Serialize tactic using pickle + hex encoding for robust handling of any Python object
+            try:
+                serialized_tactic = pickle.dumps(tactic).hex()
+            except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)